### PR TITLE
Task #163318 chore: Certificate image render and social sharing

### PR DESCRIPTION
**Pull request by tushar33**

### CODE REVIEW

File: src/components/com_tjcertificate/site/views/certificate/tmpl/default_social_sharing.php [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	{
+	?>
+		<div class="share" id="share-btn-grp">
+			<ul class="list-unstyled">
+			<?php 
+			if (in_array("facebook", $sharingOptions))
+			{
+			?>
+				 <li>
+					<a id="fb" href="https://www.facebook.com/sharer/sharer.php?u=<?php echo urlencode($this->imageUrl);?>"
```
Comment (praneettekdi) [inline src/components/com_tjcertificate/site/views/certificate/tmpl/default_social_sharing.php:82]: open this link in new window

File: src/components/com_tjcertificate/media/js/certificateImage.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+		document.body.innerHTML = originalContent;
+	},
+
+	uploadImage: function (image) {
+		var result = false;
+		var certificateId = jQuery("#certificateId").val();
+		jQuery.ajax({
+			url: Joomla.getOptions('system.paths').base + "/index.php?option=com_tjcertificate&task=certificate.uploadCertificate",
+			type: 'POST',
+			async: false,
```
Comment (praneettekdi) [inline src/components/com_tjcertificate/media/js/certificateImage.js:21]: Remove this

File: src/components/com_tjcertificate/administrator/languages/en-GB/en-GB.com_tjcertificate.ini [hunk 3]
```diff
@@ -236,3 +236,14 @@ COM_TJCERTIFICATE_SHOW_PUBLIC_PRIVATE_DESC="If 'Private' option is selected then
...
 
+; Social Sharing
+COM_TJCERTIFICATE_SOCIAL_SHARING_SETINGS="Social Sharing"
+COM_TJCERTIFICATE_SOCIAL_SHARING="Enable social sharing"
+COM_TJCERTIFICATE_SOCIAL_SHARING_DESC="Enable social sharing to share the certificate on social media"
+COM_TJCERTIFICATE_SOCIAL_SHARING_FACEBOOK="Facebook share"
+COM_TJCERTIFICATE_SOCIAL_SHARING_LINKEDIN="LinkedIn share"
+COM_TJCERTIFICATE_SOCIAL_SHARING_TWITTER="Twitter share"
+
+;Cerificate Menu option
```
Comment (praneettekdi) [inline src/components/com_tjcertificate/administrator/languages/en-GB/en-GB.com_tjcertificate.ini:247]: Check spelling

File: src/components/com_tjcertificate/media/js/certificateImage.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        document.body.innerHTML = originalContent;
+    },
+
+    uploadImage: function(image) {
+        var result = false;
+        var certificateId = jQuery("#certificateId").val();
+        jQuery.ajax({
+            url: Joomla.getOptions('system.paths').base + "/index.php?option=com_tjcertificate&task=certificate.uploadCertificate",
+            type: 'POST',
+            dataType: "text",
```
Comment (praneettekdi) [inline src/components/com_tjcertificate/media/js/certificateImage.js:21]: Is this `dataType: "text"` necessary

File: src/components/com_tjcertificate/administrator/controllers/certificate.php [hunk 5]
```diff
@@ -56,4 +56,40 @@ public function download()
...
+
+		if (!JFolder::exists(JPATH_SITE . '/media/com_tjcertificate/certificates/'))
+		{
+			JFolder::create(JPATH_SITE . '/media/com_tjcertificate/certificates');
+		}
+
+		$filePath = 'media/com_tjcertificate/certificates/';
+		$dir = JPATH_SITE . '/' . $filePath;
+
+		if (file_put_contents($dir . $filename, $canvasOutput))
```
Comment (praneettekdi) [inline src/components/com_tjcertificate/administrator/controllers/certificate.php:88]: For now check if already exist, if not then generate

File: src/components/com_tjcertificate/site/views/certificate/tmpl/default_social_sharing.php [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+use Joomla\CMS\Factory;
+$document = Factory::getDocument();
+
+$sharingOptions = $this->params->get('sharing_option');
+?>
+<!-- Social Sharing button start-->
+<?php if ($this->params->get('social_sharing'))
+{
+	if ($this->params->get('social_sharing_type') === 'addthis')
```
Comment (praneettekdi) [inline src/components/com_tjcertificate/site/views/certificate/tmpl/default_social_sharing.php:22]: Have you tested this `addthis`

**Review by praneettekdi — APPROVED** [review]
Merging in emergency

File: src/components/com_tjcertificate/media/js/certificateImage.js (hunk 4 above)
Comment (tushar33) [reply src/components/com_tjcertificate/media/js/certificateImage.js:21]: not necessary, removed

File: src/components/com_tjcertificate/media/js/certificateImage.js (hunk 2 above)
Comment (tushar33) [reply src/components/com_tjcertificate/media/js/certificateImage.js]: done

**Pull request merged**